From 4a6ecb8d0b25761ecaa2742909e47ec9a567e46a Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Tue, 24 Jan 2017 15:14:15 +0100
Subject: [PATCH 08/36] drivers: net: xgene: fix build after change_mtu
 function change

References: bsc#1019351
Patch-mainline: Never, fixes build issue due to backport

350b4e33b893 drivers: net: xgene: Add change_mtu function
requires following patch series to compile:
61e84623ace3 net: centralize net_device min/max MTU checking
a52ad514fdf3 net: deprecate eth_change_mtu, remove usage

These patches are not backported as they modify lots of drivers.
Instead, this patch fix the build and add the functionality to
xgene_change_mtu()

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/net/ethernet/apm/xgene/xgene_enet_main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/apm/xgene/xgene_enet_main.c b/drivers/net/ethernet/apm/xgene/xgene_enet_main.c
index 6c7eea8b36af..013200dca4f3 100644
--- a/drivers/net/ethernet/apm/xgene/xgene_enet_main.c
+++ b/drivers/net/ethernet/apm/xgene/xgene_enet_main.c
@@ -1505,6 +1505,9 @@ static int xgene_change_mtu(struct net_device *ndev, int new_mtu)
 	struct xgene_enet_pdata *pdata = netdev_priv(ndev);
 	int frame_size;
 
+	if (new_mtu < 68 || new_mtu > XGENE_ENET_MAX_MTU)
+		return -EINVAL;
+
 	if (!netif_running(ndev))
 		return 0;
 
@@ -1851,7 +1854,6 @@ static int xgene_enet_init_hw(struct xgene_enet_pdata *pdata)
 					    buf_pool->id, ring_id);
 	}
 
-	ndev->max_mtu = XGENE_ENET_MAX_MTU;
 	pdata->phy_speed = SPEED_UNKNOWN;
 	pdata->mac_ops->init(pdata);
 
-- 
2.11.0

